### DISCUSSION TITLE
Cannot execute code, session has been disposed. Please try restarting the Kernel.

### DISCUSSION THREAD

**Discussion by fevc08** [microsoft/vscode-jupyter#13738] [Questions and Answers]
I had this problem when I try to run the code on Jupyter and I try many things but I cannot solve it.
Visual Studio Code (1.79.2, codespaces, codespaces)
Jupyter Extension Version: 2023.5.1001582324.
Python Extension Version: 2023.10.1.
Platform: linux (x64).
Workspace folder /workspaces/realestate-datacleanup-exercise, Home = /home/vscode
17:02:57.494 [info] Start refreshing Interpreter Kernel Picker (1687280577494)
17:02:57.554 [info] Using Pylance
17:03:00.137 [warn] Failed to get activated env vars for /usr/local/bin/python in 2144ms
17:03:00.165 [info] Process Execution: /usr/local/bin/python -c "import site;print("USER_BASE_VALUE");print(site.USER_BASE);print("USER_BASE_VALUE");"
17:03:00.316 [info] Process Execution: /usr/local/bin/python -m pip list
17:03:00.982 [info] Starting Kernel startUsingPythonInterpreter, .jvsc74a57bd0949777d72b0d2535278d3dc13498b2535136f6dfe0678499012e853ee9abcab1./usr/local/python./usr/local/python.-m#ipykernel_launcher  (Python Path: /usr/local/bin/python, Unknown, 3.11.3) for '/workspaces/realestate-datacleanup-exercise/project.ipynb' (disableUI=true)
17:03:01.533 [info] End refreshing Interpreter Kernel Picker (1687280577494)
17:03:01.791 [warn] Failed to get activated env vars for /usr/local/bin/python in 713ms
17:03:01.812 [info] Process Execution: /usr/local/bin/python -c "import ipykernel; print(ipykernel.__version__); print("5dc3a68c-e34e-4080-9c3e-2a532b2ccb4d"); print(ipykernel.__file__)"
17:03:01.839 [info] Process Execution: /usr/local/bin/python -m ipykernel_launcher --ip=127.0.0.1 --stdin=9008 --control=9006 --hb=9005 --Session.signature_scheme="hmac-sha256" --Session.key=b"4159491f-b4d4-45c8-b84d-6e3d711cba7a" --shell=9007 --transport="tcp" --iopub=9009 --f=~/.local/share/jupyter/runtime/kernel-v2-41222bzZ01NLthuZ9.json
    > cwd: /workspaces/realestate-datacleanup-exercise
17:03:02.551 [info] ipykernel version & path 6.23.1, ~/.local/lib/python3.11/site-packages/ipykernel/__init__.py for /usr/local/bin/python
17:03:05.076 [warn] StdErr from Kernel Process 0.00s - Debugger warning: It seems that frozen modules are being used, which may
17:03:05.085 [warn] StdErr from Kernel Process 0.01s - make the debugger miss breakpoints. Please pass -Xfrozen_modules=off
17:03:05.086 [warn] StdErr from Kernel Process 0.00s - to python to disable frozen modules.
17:03:05.086 [warn] StdErr from Kernel Process 0.00s - Note: Debugging will proceed. Set PYDEVD_DISABLE_FILE_VALIDATION=1 to disable this validation.
17:03:05.257 [warn] StdErr from Kernel Process ~/.local/lib/python3.11/site-packages/traitlets/traitlets.py:2548: FutureWarning: Supporting extra quotes around strings is deprecated in traitlets 5.0. You can use 'hmac-sha256' instead of '"hmac-sha256"' if you require traitlets >=5.
  warn(
17:03:05.257 [warn] StdErr from Kernel Process ~/.local/lib/python3.11/site-packages/traitlets/traitlets.py:2499: FutureWarning: Supporting extra quotes around Bytes is deprecated in traitlets 5.0. Use '4159491f-b4d4-45c8-b84d-6e3d711cba7a' instead of 'b"4159491f-b4d4-45c8-b84d-6e3d711cba7a"'.
  warn(
17:03:05.570 [info] Started Kernel Python 3.11.3 (pid: 41637)
17:03:05.570 [info] Started new session 1e192086-43b5-4050-a4a0-5ba54aa245c4
17:03:05.961 [info] Process Execution: /usr/local/bin/python ~/.<username>-remote/extensions/ms-toolsai.jupyter-2023.5.1001582324-linux-x64/pythonFiles/printJupyterDataDir.py
17:04:56.730 [info] Handle Execution of Cells 2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,31,33,35,37,39,41,43,45 for /workspaces/realestate-datacleanup-exercise/project.ipynb
17:04:56.775 [info] Kernel acknowledged execution of cell 2 @ 1687280696764
17:04:58.365 [info] End cell 2 execution @ 1687280698362, started @ 1687280696764, elapsed time = 1.598s
17:04:58.376 [info] Kernel acknowledged execution of cell 4 @ 1687280698368
17:04:58.378 [info] End cell 4 execution @ 1687280698375, started @ 1687280698368, elapsed time = 0.007s
17:04:58.382 [info] Kernel acknowledged execution of cell 6 @ 1687280698380
17:04:58.446 [info] End cell 6 execution @ 1687280698433, started @ 1687280698380, elapsed time = 0.053s
17:04:58.454 [info] Kernel acknowledged execution of cell 8 @ 1687280698450
17:04:58.492 [info] End cell 8 execution @ 1687280698488, started @ 1687280698450, elapsed time = 0.038s
17:04:58.496 [info] Kernel acknowledged execution of cell 10 @ 1687280698493
17:04:58.552 [info] End cell 10 execution @ 1687280698549, started @ 1687280698493, elapsed time = 0.056s
17:04:58.557 [info] Kernel acknowledged execution of cell 12 @ 1687280698554
17:04:58.620 [info] End cell 12 execution @ 1687280698616, started @ 1687280698554, elapsed time = 0.062s
17:04:58.624 [info] Kernel acknowledged execution of cell 14 @ 1687280698621
17:04:58.708 [info] End cell 14 execution @ 1687280698706, started @ 1687280698621, elapsed time = 0.085s
17:04:58.714 [info] Kernel acknowledged execution of cell 16 @ 1687280698710
17:04:58.761 [info] End cell 16 execution @ 1687280698729, started @ 1687280698710, elapsed time = 0.019s
17:04:58.767 [info] Kernel acknowledged execution of cell 18 @ 1687280698764
17:05:04.179 [error] Disposing session as kernel process died ExitCode: undefined, Reason: 0.00s - Debugger warning: It seems that frozen modules are being used, which may
0.01s - make the debugger miss breakpoints. Please pass -Xfrozen_modules=off
0.00s - to python to disable frozen modules.
0.00s - Note: Debugging will proceed. Set PYDEVD_DISABLE_FILE_VALIDATION=1 to disable this validation.
~/.local/lib/python3.11/site-packages/traitlets/traitlets.py:2548: FutureWarning: Supporting extra quotes around strings is deprecated in traitlets 5.0. You can use 'hmac-sha256' instead of '"hmac-sha256"' if you require traitlets >=5.
  warn(
~/.local/lib/python3.11/site-packages/traitlets/traitlets.py:2499: FutureWarning: Supporting extra quotes around Bytes is deprecated in traitlets 5.0. Use '4159491f-b4d4-45c8-b84d-6e3d711cba7a' instead of 'b"4159491f-b4d4-45c8-b84d-6e3d711cba7a"'.
  warn(

17:05:04.180 [info] Dispose Kernel process 41637.
17:05:04.182 [error] Raw kernel process exited code: undefined
17:05:04.188 [error] Error in waiting for cell to complete Error: Canceled future for execute_request message before replies were done
    at t.KernelShellFutureHandler.dispose (~/.<username>-remote/extensions/ms-toolsai.jupyter-2023.5.1001582324-linux-x64/out/extension.node.js:2:32375)
    at ~/.<username>-remote/extensions/ms-toolsai.jupyter-2023.5.1001582324-linux-x64/out/extension.node.js:2:51427
    at Map.forEach (<anonymous>)
    at y._clearKernelState (~/.<username>-remote/extensions/ms-toolsai.jupyter-2023.5.1001582324-linux-x64/out/extension.node.js:2:51412)
    at y.dispose (~/.<username>-remote/extensions/ms-toolsai.jupyter-2023.5.1001582324-linux-x64/out/extension.node.js:2:44894)
    at ~/.<username>-remote/extensions/ms-toolsai.jupyter-2023.5.1001582324-linux-x64/out/extension.node.js:24:112498
    at ne (~/.<username>-remote/extensions/ms-toolsai.jupyter-2023.5.1001582324-linux-x64/out/extension.node.js:2:1586779)
    at cy.dispose (~/.<username>-remote/extensions/ms-toolsai.jupyter-2023.5.1001582324-linux-x64/out/extension.node.js:24:112474)
    at uy.dispose (~/.<username>-remote/extensions/ms-toolsai.jupyter-2023.5.1001582324-linux-x64/out/extension.node.js:24:119757)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
17:05:04.191 [warn] Cell completed with errors {
  message: 'Canceled future for execute_request message before replies were done'
}
17:05:04.195 [info] End cell 18 execution @ 1687280704194, started @ 1687280698764, elapsed time = 5.43s
17:05:04.195 [warn] Cancel all remaining cells due to cancellation or failure in execution
17:05:04.196 [info] End cell 20 execution @ undefined, started @ undefined, elapsed time = 0s
17:05:04.196 [info] End cell 22 execution @ undefined, started @ undefined, elapsed time = 0s
17:05:04.196 [info] End cell 24 execution @ undefined, started @ undefined, elapsed time = 0s
17:05:04.196 [info] End cell 26 execution @ undefined, started @ undefined, elapsed time = 0s
17:05:04.197 [info] End cell 28 execution @ undefined, started @ undefined, elapsed time = 0s
17:05:04.197 [info] End cell 30 execution @ undefined, started @ undefined, elapsed time = 0s
17:05:04.197 [info] End cell 31 execution @ undefined, started @ undefined, elapsed time = 0s
17:05:04.197 [info] End cell 33 execution @ undefined, started @ undefined, elapsed time = 0s
17:05:04.197 [info] End cell 35 execution @ undefined, started @ undefined, elapsed time = 0s
17:05:04.198 [info] End cell 37 execution @ undefined, started @ undefined, elapsed time = 0s
17:05:04.198 [info] End cell 39 execution @ undefined, started @ undefined, elapsed time = 0s
17:05:04.198 [info] End cell 41 execution @ undefined, started @ undefined, elapsed time = 0s
17:05:04.198 [info] End cell 43 execution @ undefined, started @ undefined, elapsed time = 0s
17:05:04.198 [info] End cell 45 execution @ undefined, started @ undefined, elapsed time = 0s
17:05:04.702 [info] End cell 18 execution @ undefined, started @ undefined, elapsed time = 0s

**Comment by DonJayamanne** [comment]
> End cell 16 execution @ 1687280698729, started @ 1687280698710, elapsed time = 0.019s
17:04:58.767 [info] Kernel acknowledged execution of cell 18 @ 1687280698764
17:05:04.179 [error] Disposing session as kernel process died ExitCode: undefined, Reason: 0.00s - Debugger warning: It seems that frozen modules are being used, which may

Thanks for this issue and sorry you are running into this.

Looks like your kernel crashed while running a cell, probably the 18th cell
We have seen this happens a lot when packages such as PyTorch/Tensorflow or the like are not correctly installed and causes the kernel to crash when those packages are imported/used in some form.

Please could you check what code is executed in the 18th cell, optionally please try to isolate the code that causes the kernel to crash and try to figure out what in that code/package could be causing the kernel to crash.

**Comment by fevc08** [comment]
This is the code:
data_prices = ds_nona['price']
bins = np.arange(min(data_prices), max(data_prices) + 1, 1)

x = ds_nona.loc[ds_nona['level5'] == 'Arroyomolinos (Madrid)']

plt.hist(x['price'], bins)
plt.xlabel('Frecuancy')
plt.ylabel('Prices')
plt.title('Prices of Population Arroyomolinos (Madrid)')
plt.show()

but I can't isolate it,

**Reply by DonJayamanne** [reply]
> but I can't isolate it,

Does the kernel crash if you do not run the above cell?
If not, then there's a problem with the package `numpy` or the like

You might want to try this outside VS Code or reach out to stackover flow or the like, as this is most likely not an issue with VS Code as you seem to be able to run other cells.

Sorry I cannot help you figure out what is going on.

**Reply by fevc08** [reply]
Yes, all the code above run but when I try to run that part the kernel crash.

But thank you, I will try something else
